### ISSUE TITLE
[Bug]: `ERROR 20702 (HY000): lock table bind changed` is reported when inserting record

### ISSUE THREAD

**Issue by DanielZhangQD** [matrixorigin/matrixone#12708]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):nightly-5fd8256e
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

- Connect to cn `default-cn-5sfqx` in prod cluster
- Insert record `insert into cu values ('65a970d0_9c37_41d1_9865-0f4356d85ba8', '2023-11-13 04:22:30.000', '2023-11-13 04:23:30.000', 1000, 'external_sql');` and error `ERROR 20702 (HY000): lock table bind changed` is returned

### Expected Behavior

Inserting succeeds

### Steps to Reproduce

```Markdown
See detail in Actual Behavior
```

### Additional information

No pod restarted in a long time:
```
➜  ~ kgp -n freetier-01
NAME                             READY   STATUS    RESTARTS        AGE
default-cn-5sfqx                 1/1     Running   0               7h32m
default-cn-f2qgr                 1/1     Running   0               5h45m
default-dn-0                     1/1     Running   0               6h46m
default-log-0                    1/1     Running   1 (7h31m ago)   18h
default-log-1                    1/1     Running   2 (7h31m ago)   18h
default-log-2                    1/1     Running   1 (7h32m ago)   18h
proxy-proxy-4b6jw                2/2     Running   2 (7h30m ago)   18h
proxy-proxy-rflnv                2/2     Running   2 (7h30m ago)   18h
```
Logs:
http://121.43.41.174/explore?panes=%7B%22lKn%22:%7B%22datasource%22:%22P3C839627617F91E0%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bapp%3D%5C%22default-cn%5C%22%7D%20%7C%3D%20%60%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22P3C839627617F91E0%22%7D,%22editorMode%22:%22builder%22%7D%5D,%22range%22:%7B%22from%22:%22now-6h%22,%22to%22:%22now%22%7D%7D%7D&schemaVersion=1&orgId=1

**Comment by zhangxu19830126** [comment]
fixed by #12712

**Comment by DanielZhangQD** [comment]
Verified and passed with commit `nightly-90dac68d`

**Comment by xiaoshuwei** [comment]
日志：
http://47.98.35.103/explore?panes=%7B%22LQp%22:%7B%22datasource%22:%22loki%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bapp%3D%5C%22default-cn%5C%22%7D%20%7C%3D%20%6012%2002:53:49%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki%22%7D,%22editorMode%22:%22builder%22%7D%5D,%22range%22:%7B%22from%22:%22now-6h%22,%22to%22:%22now%22%7D%7D%7D&schemaVersion=1&orgId=1

**Comment by DanielZhangQD** [comment]
Reproduced in QA env.
```
2024/01/12 02:53:49.853598 +0000 ERROR workbook/workbook.go:270 trace {"error": "Error 20702 (HY000): lock table bind changed\nPrevious DML conflicts with existing constraints or data format. This transaction has to be aborted", "elapsed": "3.772798ms", "rows": 0, "sql": "INSERT INTO `mo_cloud`.`workbook` (`id`,`account_id`,`sql_user`,`name`,`created_at`,`updated_at`) VALUES ('fad4c1a9-feff-4d91-8d8e-5a56d43337f1','3f138cf7_3c88_4269_92ab_3c4b7f1f6982','admin','default','2024-01-12 02:53:49.85','2024-01-12 02:53:49.85')"}
```
当时集群在扩容，不确定是否有关系：
<img width="1059" alt="企业微信截图_ede65b2f-94f5-4b07-82f8-4279dcefcd7c" src="https://github.com/matrixorigin/matrixone/assets/36026334/04461a9d-2c74-4ff8-90de-d331c3f776ba">

上述 SQL 打到了共享 CN，并没有打到付费实例的独享 CN：
```
select statement_id,node_uuid,statement,status,error,err_code from system.statement_info where request_at >= '2024-01-12 02:50:00' and account = 'sys' and statement like '%INSERT INTO `mo_cloud`.`workbook`%' order by request_at desc;

| f99c0a70-b0f6-11ee-8575-c22866b32e53 | 39623938-3866-6237-3737-633265643935 | /* 1 queries */
/* ApplicationName=GoLand 2023.1 */ INSERT INTO `mo_cloud`.`workbook` (`id`,`account_id`,`sql_user`,`name`,`created_at`,`updated_at`) VALUES ('fad4c1a9-feff-4d91-8d8e-5a56d43337f1','3f138cf7_3c88_4269_92ab_3c4b7f1f6982','admin','default','2024-01-12 02:53:49.85','2024-01-12 02:53:49.85')   

MySQL [(none)]> show backend servers;
+--------------------------------------+---------------------+------------+-----------------------------------------------+
| UUID                                 | Address             | Work State | Labels                                        |
+--------------------------------------+---------------------+------------+-----------------------------------------------+
| 65636663-3538-6362-3930-303166373634 | 172.20.213.132:6001 | Working    | account:sys;                                  |
| 33336434-3566-3366-3365-353832376330 | 172.20.225.131:6001 | Working    | account:3f138cf7_3c88_4269_92ab_3c4b7f1f6982; |
| 30636463-3865-6534-3761-343738636465 | 172.20.218.3:6001   | Working    | account:xzxiong;                              |
| 35613663-3431-3661-3261-386465613633 | 172.20.199.3:6001   | Working    |                                               |
| 38366639-3631-3861-3532-363265666138 | 172.20.226.131:6001 | Working    | account:3f138cf7_3c88_4269_92ab_3c4b7f1f6982; |
| 37373066-3965-6333-3761-613965373838 | 172.20.227.131:6001 | Working    | account:3f138cf7_3c88_4269_92ab_3c4b7f1f6982; |
| 32386532-3433-6238-6237-663966643930 | 172.20.217.3:6001   | Working    | account:sys;                                  |
| 66633731-3935-3837-6139-343534353730 | 172.20.222.131:6001 | Working    | account:8b0e97d1_1e94_49c8_ac18_c70192a39b3a; |
| 31623163-6361-3262-3837-616262346338 | 172.20.221.131:6001 | Working    | account:3f138cf7_3c88_4269_92ab_3c4b7f1f6982; |
| 39623938-3866-6237-3737-633265643935 | 172.20.209.2:6001   | Working    |                                               |
| 35366331-3937-6433-6662-306639623533 | 172.20.228.131:6001 | Draining   |                                               |
| 33373962-6364-3965-3331-643261346266 | 172.20.227.3:6001   | Working    | account:3f138cf7_3c88_4269_92ab_3c4b7f1f6982; |
| 65366134-6563-3439-3261-656131656666 | 172.20.225.3:6001   | Working    | account:3f138cf7_3c88_4269_92ab_3c4b7f1f6982; |
| 34363232-6539-3263-3334-303631356366 | 172.20.216.131:6001 | Working    | role:__MO_OB__;account:*;                     |
| 61626536-3633-3236-6531-626534393063 | 172.20.226.4:6001   | Working    | account:3f138cf7_3c88_4269_92ab_3c4b7f1f6982; |
| 66653137-6132-3534-3138-326262333063 | 172.20.228.3:6001   | Working    | account:3f138cf7_3c88_4269_92ab_3c4b7f1f6982; |
+--------------------------------------+---------------------+------------+-----------------------------------------------+
16 rows in set (0.00 sec)

NAME                                   READY   STATUS    RESTARTS      AGE     IP               NODE                       NOMINATED NODE   READINESS GATES
default-cn-687fx                       1/1     Running   1 (8d ago)    9d      172.20.199.3     cn-hangzhou.10.3.198.121   <none>           3/3
default-cn-vbv59                       1/1     Running   0             8d      172.20.209.2     cn-hangzhou.10.3.198.144   <none>           3/3
```

**Comment by DanielZhangQD** [comment]
上午出现了两次
```
2024/01/12 02:53:08.305391 +0000 ERROR workbook/workbook.go:270 trace {"error": "Error 20702 (HY000): lock table bind changed\nPrevious DML conflicts with existing constraints or data format. This transaction has to be aborted", "elapsed": "41.718336ms", "rows": 0, "sql": "INSERT INTO `mo_cloud`.`workbook` (`id`,`account_id`,`sql_user`,`name`,`created_at`,`updated_at`) VALUES ('16359f79-3ec3-425b-8b8e-6957b0e50237','3f138cf7_3c88_4269_92ab_3c4b7f1f6982','admin','default','2024-01-12 02:53:08.264','2024-01-12 02:53:08.264')"}
2024/01/12 02:53:08.305441 +0000 ERROR logging/api.go:59 Create workbook error: Error 20702 (HY000): lock table bind changed
2024/01/12 02:53:08.305587 +0000 DEBUG middleware/LogRequest.go:50 RespBody:{"code":"ErrInternal","msg":"Internal Server Error: Error 20702 (HY000): lock table bind changed\nPrevious DML conflicts with existing constraints or data format. This transaction has to be aborted","request_id":"08f4c6f0-ab55-4b03-bfac-cb552e7b6f95"}, RespHeaders:{{Vary:Origin}
2024/01/12 02:53:49.853598 +0000 ERROR workbook/workbook.go:270 trace {"error": "Error 20702 (HY000): lock table bind changed\nPrevious DML conflicts with existing constraints or data format. This transaction has to be aborted", "elapsed": "3.772798ms", "rows": 0, "sql": "INSERT INTO `mo_cloud`.`workbook` (`id`,`account_id`,`sql_user`,`name`,`created_at`,`updated_at`) VALUES ('fad4c1a9-feff-4d91-8d8e-5a56d43337f1','3f138cf7_3c88_4269_92ab_3c4b7f1f6982','admin','default','2024-01-12 02:53:49.85','2024-01-12 02:53:49.85')"}
2024/01/12 02:53:49.853626 +0000 ERROR logging/api.go:59 Create workbook error: Error 20702 (HY000): lock table bind changed
2024/01/12 02:53:49.853710 +0000 DEBUG middleware/LogRequest.go:50 RespBody:{"code":"ErrInternal","msg":"Internal Server Error: Error 20702 (HY000): lock table bind changed\nPrevious DML conflicts with existing constraints or data format. This transaction has to be aborted","request_id":"d260f24a-9b32-4401-8802-5e51d4f817f9"}, RespHeaders:{{Vary:Origin}
```

**Comment by iamlinjunhong** [comment]
正在处理

**Comment by iamlinjunhong** [comment]
正在处理

**Comment by iamlinjunhong** [comment]
fixed in https://github.com/matrixorigin/matrixone/pull/14296

**Comment by DanielZhangQD** [comment]
@iamlinjunhong Fix 进到 1.1-dev 了吗?

**Comment by aressu1985** [comment]
Fixed
